buster/documents_manager/base.py

```diff
@@ -185,12 +185,15 @@ def batch_add(
             )
 
             elapsed_time = time.time() - start_time
-            sleep_time = max(0, min_time_interval - elapsed_time)
+
 
             # Sleep to ensure the minimum time interval is maintained
-            if sleep_time > 0:
-                logger.info(f"Sleeping for {round(sleep_time)} seconds...")
-                time.sleep(sleep_time)
+            # Only sleep if it's not the last iteration
+            if batch_idx < total_batches - 1:
+                sleep_time = max(0, min_time_interval - elapsed_time)
+                if sleep_time > 0:
+                    logger.info(f"Sleeping for {round(sleep_time)} seconds...")
```

Comment:
I would move this call outside the if, and have it print the elapsed time instead. Just so we get an idea of average batch processing time.

Reply:
Yes the logger call. I guess tqdm might be enough.